Orange/widgets/visualize/owvenndiagram.py

```diff
@@ -361,17 +362,17 @@ def merge_data(self, domain, values, ids=None):
         for val in domain.values():
             names = [var.name for var in val]
             unique_names = get_unique_names_duplicates(names)
-            for n, u, idx, var in zip(names, unique_names, range(len(val)), val):
+            for n, u, idx, var in zip(names, unique_names, count(), val):
                 if n != u:
                     val[idx] = var.copy(name=u)
                     renamed.append(n)
         if renamed:
             self.Warning.renamed_vars(', '.join(renamed))
-        if 'attributes' in values.keys():
+        if 'attributes' in values:
```

Comment:
I thing skipping `keys` is alse faster here -- `... in values.keys()` could have O(n) complexity.

Reply:
Uf. Yes, looking in values.keys() would defeat the whole purpose of hashmaps.